### PR TITLE
FEATURE: Make site texts controller handle pages and locales

### PR DESCRIPTION
**Pull request by nbianca**

### CODE REVIEW

File: app/controllers/admin/site_texts_controller.rb [hunk 1]
```diff
@@ -24,7 +24,11 @@ def index
       extras[:recommended] = true
       results = self.class.preferred_keys.map { |k| record_for(k) }
     else
-      results = find_translations(query, overridden)
+      if locale = params[:locale].presence
+        results = I18n.with_locale(locale) { find_translations(query, overridden) }
```
Comment (ZogStriP) [inline app/controllers/admin/site_texts_controller.rb]: What happens if `params[:locale]` is some gibberish?

File: app/controllers/admin/site_texts_controller.rb [hunk 2]
```diff
@@ -41,8 +45,13 @@ def index
       end
     end
 
-    extras[:has_more] = true if results.size > 50
-    render_serialized(results[0..49], SiteTextSerializer, root: 'site_texts', rest_serializer: true, extras: extras, overridden_keys: overridden_keys)
+    per_page = 50
+    page = params[:page].to_i
```
Comment (ZogStriP) [inline app/controllers/admin/site_texts_controller.rb]: What if `params[:page]` is `-99`?

File: spec/requests/admin/site_texts_controller_spec.rb [hunk 3]
```diff
@@ -58,6 +58,28 @@
...
+
+        get "/admin/customize/site_texts.json", params: { q: 'e' }
+        JSON.parse(response.body)['site_texts'].each { |text| texts << text['id'] }
+        expect(texts.size).to eq(100)
+      end
+
+      it 'works with locales' do
+        get "/admin/customize/site_texts.json", params: { q: 'yes_value', locale: 'en' }
+        value = obj['site_texts'].find { |text| text['id'] == 'js.yes_value' }['value']
+        expect(value).to eq('Yes')
```
Comment (ZogStriP) [inline spec/requests/admin/site_texts_controller_spec.rb]: This test will fail if/when the translation changes. It's best to call `I18n.t()` with the `en` locale to make this test less fragile.

File: spec/requests/admin/site_texts_controller_spec.rb [hunk 4]
```diff
@@ -58,6 +58,28 @@
...
+
+        get "/admin/customize/site_texts.json", params: { q: 'yes_value', locale: 'de' }
+        value = obj['site_texts'].find { |text| text['id'] == 'js.yes_value' }['value']
+        expect(value).to eq('Ja')
```
Comment (ZogStriP) [inline spec/requests/admin/site_texts_controller_spec.rb]: Ditto about the test "fragility"

**Comment by ZogStriP** [comment]
Why is this change needed? Can you add a description explaining why the change and how it solves the problem?

File: app/controllers/admin/site_texts_controller.rb [hunk 5]
```diff
@@ -41,8 +45,15 @@ def index
...
+    first = page * per_page
+    last = first + per_page
+
+    raise Discourse::InvalidParameters.new(:page) if page < 0
```
Comment (ZogStriP) [inline app/controllers/admin/site_texts_controller.rb]: If `page < 0`, then you have calculated `first` and `last` for nothing 😉

Comment (nbianca) [reply app/controllers/admin/site_texts_controller.rb]: I wanted to keep all computations in one place hoping to make things more clear. 😊

File: app/controllers/admin/site_texts_controller.rb [hunk 6]
```diff
@@ -41,8 +45,16 @@ def index
       end
     end
 
-    extras[:has_more] = true if results.size > 50
-    render_serialized(results[0..49], SiteTextSerializer, root: 'site_texts', rest_serializer: true, extras: extras, overridden_keys: overridden_keys)
+    per_page = 50
```
Comment (ZogStriP) [inline app/controllers/admin/site_texts_controller.rb]: This one could also be moved near where it's being used after the `raise` 😉

**Pull request merged**